### PR TITLE
Variable scope handling 

### PR DESCRIPTION
**Pull request by erandacr**

### CODE REVIEW

File: gateway-core/components/org.wso2.carbon.gateway.core/src/main/java/org/wso2/carbon/gateway/core/config/dsl/external/wuml/WUMLBaseListenerImpl.java [hunk 1]
```diff
@@ -608,8 +599,15 @@ public void exitInvokeMediatorCall(WUMLParser.InvokeMediatorCallContext ctx) {
 
         ParameterHolder parameterHolder = new ParameterHolder();
         parameterHolder.addParameter(new Parameter("endpointKey", ctx.Identifier().get(0).getText()));
+        parameterHolder.addParameter(new Parameter("inputMessageParameter", ctx.Identifier().get(1).getText()));
```
Comment (vinok88) [inline gateway-core/components/org.wso2.carbon.gateway.core/src/main/java/org/wso2/carbon/gateway/core/config/dsl/external/wuml/WUMLBaseListenerImpl.java]: Shouldn't these string go to constants?

File: gateway-core/components/org.wso2.carbon.gateway.core/src/test/java/org/wso2/carbon/gateway/core/exceptions/ExceptionTest.java [hunk 2]
```diff
@@ -154,6 +161,7 @@ public void mediatorAfterTryBlockMediatorTest() throws Exception {
         resource.getDefaultWorker().addMediator(tryBlockMediator);
         resource.getDefaultWorker().addMediator(outSideCatch);
 
+        resource.setInputParamIdentifier("some_value");
```
Comment (nandika) [inline gateway-core/components/org.wso2.carbon.gateway.core/src/test/java/org/wso2/carbon/gateway/core/exceptions/ExceptionTest.java]: can you explain the usage of this parameter "some_value"

File: gateway-core/components/org.wso2.carbon.gateway.core/src/main/java/org/wso2/carbon/gateway/core/config/dsl/external/wuml/WUMLBaseListenerImpl.java [hunk 3]
```diff
@@ -638,27 +636,42 @@ public void exitLogMediatorCall(WUMLParser.LogMediatorCallContext ctx) {
         dropMediatorFilterAware(logMediator);
     }
 
+    /** statements with 'reply'. It maps to a Respond mediator*/
+    @Override public void exitReturnStatement(WUMLParser.ReturnStatementContext ctx) {
+        Mediator respondMediator = MediatorProviderRegistry.getInstance().getMediator("respond");
+        ParameterHolder parameterHolder = new ParameterHolder();
+        if (ctx.Identifier() != null) {
+            String messageId = ctx.Identifier().getText();
+            parameterHolder.addParameter(new Parameter("messageId", messageId));
```
Comment (VijithaEkanayake) [inline gateway-core/components/org.wso2.carbon.gateway.core/src/main/java/org/wso2/carbon/gateway/core/config/dsl/external/wuml/WUMLBaseListenerImpl.java]: Shall we define constants for frequently use strings like "messageId", "key", "type", etc...

File: gateway-core/components/org.wso2.carbon.gateway.core/src/test/java/org/wso2/carbon/gateway/core/exceptions/ExceptionTest.java (hunk 2 above)
Comment (erandacr) [inline gateway-core/components/org.wso2.carbon.gateway.core/src/test/java/org/wso2/carbon/gateway/core/exceptions/ExceptionTest.java]: For a resource, we are passing a message with Identifier. This value is set in the resource object when parsing the configuration. [1] 
There for here, "some_value" is referred to the 'message' identifier. 

[1]. https://github.com/wso2/carbon-gateway-framework/pull/43/files#diff-13c7c7846b9fede8be5f29d09a00c751R420

File: gateway-core/components/org.wso2.carbon.gateway.core/src/main/java/org/wso2/carbon/gateway/core/config/dsl/external/wuml/WUMLBaseListenerImpl.java (hunk 1 above)
Comment (erandacr) [inline gateway-core/components/org.wso2.carbon.gateway.core/src/main/java/org/wso2/carbon/gateway/core/config/dsl/external/wuml/WUMLBaseListenerImpl.java]: yeah, I will add contents for frequently used strings.

File: gateway-core/components/org.wso2.carbon.gateway.core/src/main/java/org/wso2/carbon/gateway/core/config/dsl/external/wuml/WUMLBaseListenerImpl.java (hunk 3 above)
Comment (erandacr) [inline gateway-core/components/org.wso2.carbon.gateway.core/src/main/java/org/wso2/carbon/gateway/core/config/dsl/external/wuml/WUMLBaseListenerImpl.java]: yeah, I will add contents for frequently used strings.

File: gateway-core/components/org.wso2.carbon.gateway.core/src/test/java/org/wso2/carbon/gateway/core/exceptions/ExceptionTest.java (hunk 2 above)
Comment (erandacr) [inline gateway-core/components/org.wso2.carbon.gateway.core/src/test/java/org/wso2/carbon/gateway/core/exceptions/ExceptionTest.java]: I will give a meaning full text to that value

Comment (nandika) [inline gateway-core/components/org.wso2.carbon.gateway.core/src/test/java/org/wso2/carbon/gateway/core/exceptions/ExceptionTest.java]: let's add a meaningful text

**Pull request merged**